### PR TITLE
Add referral network parameters

### LINKED ISSUE #8914: Add referral network parameters
Add referral network parameters and wire them to the referral engine.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #8914

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,7 @@
 - [8895](https://github.com/vegaprotocol/vega/issues/8895) - Allow to set runtime parameters in the SQL Store connection structure
 - [8779](https://github.com/vegaprotocol/vega/issues/8779) - Query all details of liquidity providers via an API.
 - [8924](https://github.com/vegaprotocol/vega/issues/8924) - Refactor slightly to remove need to deep clone `proto` types
-- [8782](https://github.com/vegaprotocol/vega/issues/8782) - List all active liquidity providers fora  market via API.
+- [8782](https://github.com/vegaprotocol/vega/issues/8782) - List all active liquidity providers fora market via API.
```
Comment (peterbarrow) [inline CHANGELOG.md:32]:
```suggestion
- [8782](https://github.com/vegaprotocol/vega/issues/8782) - List all active liquidity providers for a market via API.
```

**Review by peterbarrow — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

**Review by jeremyletang — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

File: core/netparams/defaults.go [hunk 2]
```diff
@@ -257,6 +257,13 @@ func defaultNetParams() map[string]value {
 
 		RewardsVestingBaseRate:       NewDecimal(gtD0, lteD1).Mutable(true).MustUpdate("0.01"),
 		RewardVestingMinimumTransfer: NewDecimal(gtD0, lteD1).Mutable(true).MustUpdate("0.01"),
+
+		// Referral program
+		ReferralProgramMaxBenefitTiers:           NewUint(UintGTE(num.NewUint(0)), UintLTE(num.NewUint(100))).Mutable(true).MustUpdate("0"),
+		ReferralProgramMaxReferralRewardFactor:   NewDecimal(gteD0, lteD100).Mutable(true).MustUpdate("0"),
+		ReferralProgramMaxReferralDiscountFactor: NewDecimal(gteD0, lteD100).Mutable(true).MustUpdate("0"),
+		ReferralProgramMaxPartyVolumePerEpoch:    NewUint(UintGTE(num.NewUint(0)), UintLTE(num.NewUint(10000000))).Mutable(true).MustUpdate("0"),
+		ReferralProgramMinStakedVegaTokens:       NewDecimal(gteD1).Mutable(true).MustUpdate("1000000000000000000"),
```
Comment (davidsiska-vega) [inline core/netparams/defaults.go:266]: can we write in a notation that doesn't have me counting zeros? Something `num.MustDecimalFromString("1e18")`?

**Review by davidsiska-vega — CHANGES_REQUESTED** [review]

File: core/netparams/defaults.go (hunk 2 above)
Comment (ValentinTrinque) [reply core/netparams/defaults.go]: Done.

**Review by jeremyletang — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

**Review by cdummett — DISMISSED** [review]

**Review by davidsiska-vega — CHANGES_REQUESTED** [review]

**Review by davidsiska-vega — APPROVED** [review]
I *think* we're good now, @cdummett / @barnabee pls check too. And yes, maybe we want to set the right defaults now but I'd be happy to do that as a separate PR - otherwise we'll be blocking.

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**